File: Level1/Materal.Workflow/Materal.Workflow.Vue/src/scripts/StepDatas/IfStepData.ts
import { DecisionConditionData } from "./Base/DecisionConditionData";
import { IStepData } from "./Base/IStepData";
import { StepData } from "./Base/StepData";

export class IfStepData extends StepData {
    /**
     * 决策条件数据
     */
    public Conditions: DecisionConditionData[] = []
    /**
     * 计划节点
     */
    public StepData?: IStepData;
    /**
     * 下一步
     */
    public Next?: IStepData;
    constructor(id: string) {
        super(`${IfStepData.name}`, "决策节点", id);
    }
}

File: Level2/Materal.Workflow/Materal.Workflow/StepDatas/IfStep/IfStepData.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Materal.Workflow.StepDatas
{
    /// <summary>
    /// 条件节点数据
    /// </summary>
    public class IfStepData : StepData, IStepData
    {
        /// <summary>
        /// 决策条件数据
        /// </summary>
        [Required, MinLength(1)]
        public List<DecisionConditionData> Conditions { get; set; } = new();
        /// <summary>
        /// 满足条件执行的节点数据
        /// </summary>
        [Required]
        public IStepData StepData { get; set; } = new StartStepData();
        /// <summary>
        /// 下一步
        /// </summary>
        [System.Text.Json.Serialization.JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public IStepData? Next { get; set; }
        /// <summary>
        /// 构造方法
        /// </summary>
        public IfStepData()
        {

        }
        /// <summary>
        /// 构造方法
        /// </summary>
        /// <param name="leftValue"></param>
        /// <param name="leftValueSource"></param>
        /// <param name="comparisonType"></param>
        /// <param name="rightValue"></param>
        /// <param name="rightValueSource"></param>
        public IfStepData(object leftValue, ValueSourceEnum leftValueSource, ComparisonTypeEnum comparisonType, object rightValue, ValueSourceEnum rightValueSource)
        {
            Conditions.Add(new(leftValue, leftValueSource, comparisonType, rightValue, rightValueSource));
        }
    }
}
